### PR TITLE
Fix performance issue when selecting items with arrow keys

### LINKED ISSUE #8: Performance issue
When a row is selected, the SelectionGridPro fetches the model again from the data provider, despite it already being available in the grid. This results in an excessive number of database calls, especially when users continuously press the arrow key, severely impacting the application's performance.
Instead of maintaining the selected models, it would be more efficient to maintain the selected index in the grid. By invoking the data provider or fetching from the grid cache whenever the getSelectedItems method is called, by the way unnecessary calls to the data provider can be reduced.

### PR DESCRIPTION
**Pull request by mlopezFC**

### CODE REVIEW

**Comment by javier-godoy** [comment]
The suggested approach improves the performance when the requested item is already present in the key mapper. However, if the user selects a row, scrolls down and then selects another row, it will result in N separate fetches instead of a single one.

File: selection-grid-pro-flow/src/main/java/com/vaadin/componentfactory/selectiongridpro/SelectionGridPro.java [hunk 1]
```diff
@@ -148,15 +148,12 @@ private void selectRange(int fromIndex, int toIndex) {
...
-            Method fetchFromProvider;
-            try {
-                fetchFromProvider = DataCommunicator.class.getDeclaredMethod("fetchFromProvider", int.class, int.class);
-                fetchFromProvider.setAccessible(true);
-                asMultiSelect().select(((Stream<T>) fetchFromProvider.invoke(dataCommunicator, Math.min(fromIndex, toIndex), Math.max(fromIndex,
-                        toIndex) - Math.min(fromIndex, toIndex) + 1)).collect(Collectors.toList()));
-            } catch (Exception ignored) {
-                ignored.printStackTrace();
+            Set<T> newSelectedItems = new HashSet<>();
+            for(int i = Math.min(fromIndex, toIndex); i < (Math.max(fromIndex,
```
Comment (javier-godoy) [inline selection-grid-pro-flow/src/main/java/com/vaadin/componentfactory/selectiongridpro/SelectionGridPro.java:152]: There is no need to compute this value in each iteration.

Comment (mlopezFC) [reply selection-grid-pro-flow/src/main/java/com/vaadin/componentfactory/selectiongridpro/SelectionGridPro.java]: Fixed ✔️

File: selection-grid-pro-flow/src/main/java/com/vaadin/componentfactory/selectiongridpro/SelectionGridPro.java [hunk 2]
```diff
@@ -147,19 +147,35 @@ private String getColumnInternalId(Column<T> column) {
...
     }
 
+	@SuppressWarnings("unchecked")
+	private Set<T> obtainNewSelectedItems(int fromIndex, int toIndex) {
+		DataCommunicator<T> dataCommunicator = super.getDataCommunicator();
+		Set<T> newSelectedItems = new HashSet<>();
+		int from = Math.min(fromIndex, toIndex);
+		int to = Math.max(fromIndex, toIndex) + 1;
+		int pageSize = dataCommunicator.getPageSize();
+		if (to - from < (pageSize * 2) - 3) {
```
Comment (javier-godoy) [inline selection-grid-pro-flow/src/main/java/com/vaadin/componentfactory/selectiongridpro/SelectionGridPro.java:162]: Please add a comment explaining the reasoning behind this condition.

**Review by javier-godoy — CHANGES_REQUESTED** [review]

File: selection-grid-pro-flow/src/main/java/com/vaadin/componentfactory/selectiongridpro/SelectionGridPro.java (hunk 2 above)
Comment (mlopezFC) [reply selection-grid-pro-flow/src/main/java/com/vaadin/componentfactory/selectiongridpro/SelectionGridPro.java:162]: Done ✔️

**Review by javier-godoy — APPROVED** [review]

**Pull request merged**